app/src/main/java/ca/ualberta/CMPUT3012019T02/alexandria/model/Book.java

```diff
@@ -18,112 +24,195 @@
+        setTitle(title);
+        setAuthor(author);
+        setDate(date);
+        setDescription(description);
 
         availableOwners = new ArrayList<>();
     }
 
+    /**
+     * Gets the isbn of this book
+     * @return
+     */
     public String getIsbn() {
         return isbn;
     }
 
-    public void setIsbn(String isbn) {
+    /**
+     * Sets the isbn of this book
+     * @param isbn
+     */
+    public void setIsbn(@NonNull String isbn) {
         if (isbn == null || isbn.trim().isEmpty()) {
             throw new IllegalArgumentException("Isbn cannot be null or empty");
         }
 
         this.isbn = isbn;
     }
 
-    public String getImage() {
+    /**
+     * Gets the cover image id of this book
+     * @return image id
+     */
+    @Nullable
+    public String getImageId() {
         return imageId;
     }
 
-    public void setImage(String imageId) {
+    /**
+     * Sets the cover image id of this book
+     * @param imageId image id to set to
+     */
+    public void setImageId(@Nullable String imageId) {
         this.imageId = imageId;
     }
 
+    /**
+     * Gets the title of this book
+     * @return the title of the book
+     */
     public String getTitle() {
         return title;
     }
 
-    public void setTitle(String title) {
+    /**
+     * Sets the title of this book
+     * @param title the title to set
+     */
+    public void setTitle(@NonNull String title) {
         if (title == null || title.trim().isEmpty()) {
             throw new IllegalArgumentException("Title cannot be null or empty");
         }
 
         this.title = title;
     }
 
+    /**
+     * Gets the author of this book
+     * @return the name of the author
+     */
     public String getAuthor() {
         return author;
     }
 
-    public void setAuthor(String author) {
+    /**
+     * Sets the author of this book
+     * @param author the name of the author
+     */
+    public void setAuthor(@NonNull String author) {
         if (author == null || author.trim().isEmpty()) {
             throw new IllegalArgumentException("Author cannot be null or empty");
         }
 
         this.author = author;
     }
 
+    /**
+     * Gets the date of this book
+     * @return the date of this book
+     */
+    @Nullable
     public Date getDate() {
         return date;
     }
 
-    public void setDate(Date date) {
+    /**
+     * Sets the date of this book
+     * @param date date of this book
+     */
+    public void setDate(@Nullable Date date) {
         this.date = date;
     }
 
+    /**
+     * Gets a list of owner user ids that have this book available for borrowing
+     * @return a list of user ids
+     */
     public List<String> getAvailableOwners() {
         return Collections.unmodifiableList(availableOwners);
     }
 
+    /**
+     * Sets the list of available owners
+     * TO BE USED BY BOOK CONTROLLER ONLY
+     * @param availableOwners list of available owners
+     */
+    @Deprecated
```

Comment:
If it is still in use I don't think it should be @Deprecated

Reply:
"TO BE USED BY BOOK CONTROLLER ONLY" is good though

Reply:
No other class (except unit tests) should use this method, otherwise the contents of `availableOwners` can become ambiguous.

Reply:
I have not seen @Deprecated used in this way, but it makes it more clear that people should not use it so should be fine then

Reply:
I don't know of another annotation that'll get IDEs to warn the programmer about the usage of the function.